core/src/main/java/com/sequenceiq/cloudbreak/service/organization/OrganizationService.java

```diff
@@ -268,30 +273,45 @@ private void validateNoUserIsInTheOrganization(Organization organization, Set<Us
         }
     }
 
-    public Organization delete(Long id) {
+    public Organization delete(Long id, IdentityUser identityUser) {
         try {
             return transactionService.required(() -> {
-                Organization organization = get(id);
-                organizationRepository.delete(organization);
-                return organization;
+                Organization organizationForDelete = get(id);
```

Comment:
delete and deleteByName only differs in this and L297. This code duplication should be resolved.